src/chains/ethereum/transaction/src/raw.ts

```diff
@@ -1,18 +1,54 @@
+import { AccessListBuffer } from "@ethereumjs/tx";
 /**
  * The raw data for an ethereum transaction.
  */
-export type EthereumRawTx = [
+
+type Concat<T extends unknown[], U extends unknown[]> = [...T, ...U];
+
+type TxType = [type: Buffer];
+
+export type LegacyDatabasePayload = [
+  nonce: Buffer,
+  gasPrice: Buffer,
+  gas: Buffer,
+  to: Buffer,
+  value: Buffer,
+  data: Buffer,
+  v: Buffer,
+  r: Buffer,
+  s: Buffer
+];
+
+export type EIP2930AccessListDatabasePayload = [
+  chainId: Buffer,
   nonce: Buffer,
   gasPrice: Buffer,
   gas: Buffer,
   to: Buffer,
   value: Buffer,
   data: Buffer,
+  accessList: AccessListBuffer,
   v: Buffer,
   r: Buffer,
   s: Buffer
 ];
 
+export type LegacyDatabaseTx = Concat<TxType, LegacyDatabasePayload>;
+export type EIP2930AccessListDatabaseTx = Concat<
+  TxType,
+  EIP2930AccessListDatabasePayload
+>;
+
+export type TypedDatabaseTransaction =
+  | LegacyDatabaseTx
+  | EIP2930AccessListDatabaseTx;
+export type TypedDatabasePayload =
+  | LegacyDatabasePayload
+  | EIP2930AccessListDatabasePayload;
+export type BlockDatabaseTransaction =
+  | LegacyDatabasePayload
+  | EIP2930AccessListDatabaseTx;
+
 /**
  * Extra data Ganache stores as part of a transaction in order to support
  * account mascarading and quick lookups for transactions, blocks, and receipts.
```

Comment:
let's sneak this typo fix in this PR :-)

```suggestion
 * account masquerading and quick lookups for transactions, blocks, and receipts.
```